distribution/pull_v2.go

```diff
@@ -877,6 +884,31 @@ func (p *v2Puller) pullSchema2Config(ctx context.Context, dgst digest.Digest) (c
 	return configJSON, nil
 }
 
+func retry(ctx context.Context, maxAttempts int, sleep time.Duration, f func(ctx context.Context) error) (err error) {
+	for attempt := 0; attempt < maxAttempts; attempt++ {
+		err = retryOnError(f(ctx))
+		if err == nil {
+			return nil
+		}
+		if xfer.IsDoNotRetryError(err) {
+			return err
+		}
+
+		if attempt+1 < maxAttempts {
+			timer := time.NewTicker(sleep)
+			select {
+			case <-ctx.Done():
+				timer.Stop()
+				return ctx.Err()
+			case <-timer.C:
+				logrus.WithError(err).WithField("attempts", attempt+1).Error("retrying after error")
```

Comment:
Oh, perhaps got lost in the comments; could you make this a `.Debug()` (if we're retrying, it's not an actual "error"). For the "non-debug" situation we'd already return (and log) the error after we give up, so probably not strictly needed.

Reply:
Whoops, yep, missed that